telegram/bot.py

```diff
@@ -1636,8 +1664,42 @@ def answer_inline_query(self,
                     else:
                         res.input_message_content.disable_web_page_preview = None
 
-        effective_results = [res.to_dict() for res in effective_results]
-        data = {'inline_query_id': inline_query_id, 'results': effective_results}
+        if current_offset is not None and next_offset is not None:
+            raise ValueError('`current_offset` and `next_offset` are mutually exclusive!')
+
+        if current_offset is not None:
+            if current_offset == '':
+                current_offset_int = 0
+            else:
+                current_offset_int = int(current_offset)
+
+            next_offset = ''
+
+            if callable(results):
+                effective_results = results(current_offset_int)
+                if not effective_results:
+                    effective_results = []
+                else:
+                    next_offset = str(current_offset_int + 1)
+            else:
+                if len(results) > (current_offset_int + 1) * MAX_INLINE_QUERY_RESULTS:
+                    next_offset_int = current_offset_int + 1
+                    next_offset = str(next_offset_int)
+                    effective_results = results[
+                        current_offset_int * MAX_INLINE_QUERY_RESULTS:
+                        next_offset_int * MAX_INLINE_QUERY_RESULTS]
+                else:
+                    effective_results = results[current_offset_int * MAX_INLINE_QUERY_RESULTS:]
+        else:
+            effective_results = results
+
+        for result in effective_results:
+            _set_defaults(result)
+
+        results_dicts = [res.to_dict() for res in effective_results]
```

Comment:
No, it shouldn't … But it will be gone, after I rebase one more time before I merge …

Reply:
uh, no, that's just an ugly diff because I moved the setting of the defaults to the local function. noam would probably scold me for it but imho it doesn't really matter with this PR xD

Reply:
Not only noam *looks at you disapprovingly*